index.d.ts

```diff
@@ -0,0 +1,18 @@
+declare interface options {
+  ttl?: number;
+  shouldPersist?: boolean;
+}
+
+declare interface pandora<T> {
+  set<K extends keyof T>(key: K, value: T[K], opts?: options): void;
+
+  get<K extends keyof T>(key: K, defaultValue?: T[K]): T[K];
+
+  remove<K extends keyof T>(key: K): void;
+
+  clear(removeAll?: boolean): T;
```

Comment:
`clear` only returns `T` if `removeAll` is `true` or else it should be `Partial<T>`

Reply:
Same would go for getSnapshot also right? There is a possibility that some keys are yet to be written to localstorage